spread-shellcheck

```diff
@@ -101,15 +102,34 @@ class ShellcheckFailures(Exception):
         return iter(self.failures)
 
 
-def checksection(data):
+def checksection(data, env: Dict[str, str]):
     # spread shell snippets are executed under 'set -e' shell, make sure
     # shellcheck knows about that
-    data = 'set -e\n' + data
+    script_data = []
+    script_data.append('set -e')
+    for key, value in env.items():
+        value = str(value)
+        # Unpack the special "$(HOST: ...) syntax and tell shellcheck not to
+        # worry about the use of echo to print variable value.
+        if value.startswith("$(HOST:") and value.endswith(")"):
```

Comment:
+0.85 on this matching and magling, seems a bit delicate. Can we just disable SC2116 for entire environment block unconditionally? This shellcheck warning seems mildly useful in environment block anyway, no?

Reply:
About delicate: yeah, if we had spread support for emitting the effective script for a given test, we'd not have to use this hackery.

About SC2116, I don't know of a way to disable warnings for the entire block and I also don't know what you mean by block here. It's all one long shell script, with environment emitted up front.

The problem about this warning is that `spread.yaml` is coded to use `echo "$FOO"` because those have to be executable commands. When they are placed inside `$(...)` and used just for assignment then spread-shellcheck effectively sees, in many cases, `foo="$(echo "$bar")"` and says this can be simplified to just `foo="$bar"`, rightfully so.

This however is just an artefact of how we use `$(HOST: ...)` syntax and perform spellchecking **and** is not universally true because some parts of our code use actual logic to decide what to print.

Reply:
Ok, fair enough, thanks for explanation. What I meant by block was all the environment we iterate and output within this loop. Fair point with $(HOST:..) vs everything else.